xdsl/dialects/snitch_stream.py

```diff
@@ -55,58 +50,37 @@ def print_parameter(self, printer: Printer):
 
 
 @irdl_op_definition
-class GenericOp(IRDLOperation):
+class StreamingRegionOp(IRDLOperation):
```

Comment:
Can this be documented a bit better? Specifically the `inputs` and `outputs`.
At this point in the dialect we are pretty much expecting integer registers holding a memory location pointer, right?

Moreover, can you please state this separation of responsibilities (iteration and streaming) and where it happens as you have so nicely written in the PR message?

This is minor, but it made me think and had to look at the lowering to make sure that this is what is happening.

Reply:
Very good points, I'll add some documentation